core/arch/mangle_shared.c

```diff
@@ -1028,8 +1058,14 @@ mangle_rseq_insert_native_sequence(dcontext_t *dcontext, instrlist_t *ilist,
 #    endif
     PRE(ilist, insert_at, abort_sig);
     PRE(ilist, insert_at, label_abort);
-    instrlist_preinsert(ilist, insert_at,
-                        XINST_CREATE_jump(dcontext, opnd_create_pc(handler)));
+    /* To raise a kernel xfer event we need to go back to DR.  Thus this exit will
+     * never be linked.  This should be quite rare, however, and should not impose
+     * a performance burden.
```

Comment:
Doesn't it happen on every rseq exit? rseq exits are only rare if rseq sections are not super hot code, and internally they are at least somewhat hot. So "quite rare" doesn't seem like an accurate description.

Reply:
No this is only an *abort* in the native rseq run: so only on a CPU migration, context switch, or signal.  This is not on every rseq execution.  I had to work hard to get this to happen in our tests.

Reply:
Do you have a suggestion for tweaking this or prior comments to make this clearer?  Or should I just resolve.

Reply:
I was definitely just confused about the context of these changes. Your explanation in the other review thread helped me understand, thanks.